src/runtime_src/driver/xclng/drm/xocl/subdev/str_qdma.c

```diff
@@ -339,29 +342,58 @@ static ssize_t stream_post_bo(struct str_device *sdev,
 	drm_gem_object_reference(gem_obj);
 	xobj = to_xocl_bo(gem_obj);
 
-	memset(&wr, 0, sizeof (wr));
-	wr.write = write;
-	wr.len = len;
-	wr.sgt = xobj->sgt;
-	wr.eot = (header->flags & XOCL_QDMA_REQ_FLAG_EOT) ? true : false;
+	io_req = kmem_cache_alloc(req_cache, GFP_KERNEL);
```

Comment:
I am not sure how efficient kmem_cache_alloc is. And this is critical path. Should we avoid kmem_cache_alloc() on critical path? The stream_async_req structure should be able to be pre-allocated during queue creation.